enhancements/20250203-third-party-storage-support.md

```diff
@@ -0,0 +1,98 @@
+# Third Party Storage Support
+
+Third-party storage support is a feature that allows users to use third-party storage solutions in Harvester for root disks and data disks.
+
+## Summary
+
+This enhancement aims to provide a generic way to leverage third-party storage solutions which already have CSI driver. By introducing Containerized-Data Importer (CDI), Harvester can support third-party storage solutions with the current Virtual Machine Image path for the root and data disks.
```

Comment:
AIUI, CDI is just an [enhanced volume populator](https://github.com/kubernetes-csi/lib-volume-populator). Theoretically, there is nothing stopping a user from creating a pre-populated PVC using whatever image data population mechanism their CSI supports, right? In a future enhancement, is it possible to support something like  `sourceType: predefined` in the image, to allow user to "bring their own root disk"?

Reply:
Currently, user can convert their root disk to qcow2 format and upload it as our VM image.
That might be the same purpose. WDYT?

We could try to allow them to directly import their root disk. Actually, CDI already has that capability (REF: https://github.com/kubevirt/containerized-data-importer/blob/main/doc/supported_operations.md)
we need to check how to integrate and the UI flow.